agent/installer/installer.go

```diff
@@ -85,6 +91,53 @@ func (bd *bundleDownloader) DownloadOrPreview(os, k8s, tag string) error {
 	return bd.Download(os, k8s, tag)
 }
 
+func supportedOSversion(os string) error {
```

Comment:
Have you looked at https://github.com/vmware-tanzu/cluster-api-provider-bringyourownhost/blob/f285e157aa01b501aa7c54a48374c1d25d7c772d/agent/installer/installer.go#L116-L118

Looks like this is also kind of validating what OS is supported. Maybe we can leverage this? Or have only one place where we check this (i.e combine both code)

Reply:
cc @jeuxdemains

Reply:
Looking at it now

Reply:
Ok, with this we don't need another OS precheck. Hence removed the check that I had added.